composer/utils/data_helpers.py

```diff
@@ -0,0 +1,38 @@
+from composer.core.data_spec import DataSpec
+from torch.utils.data import DataLoader, Dataset
+from typing import TYPE_CHECKING, Any, Callable, Dict, Iterable, List, Optional, Sequence, Union, cast
+
+if TYPE_CHECKING:
+    from composer.core.evaluator import Evaluator
+
+__all__ = ['_dataset_of']
+
+def _dataset_of(self, dataloader: Optional[Union[Evaluator, DataSpec, DataLoader, Iterable]]) -> Optional[Dataset]:
```

Comment:
should we just put in misc since its only used in one place? I would prefer not to add more files if its only for one fn / not enough to group.

Also should this be a public fn?